samples/snippets/cpp/VS_Snippets_CLR_System/system.Func~2/cpp/Example.cpp

```diff
@@ -0,0 +1,44 @@
+using namespace System;
+using namespace System::Linq;
+using namespace System::Collections;
+using namespace System::Collections::Generic;
+
+// <Snippet6>
+// Declare a delegate
+delegate String ^ MyDel(String ^);
+
+// Create wrapper class and function that takes in a string and converts it to uppercase
+ref class A {
+public:
+    String ^ capitalize(String ^ s) {
+        return s->ToUpper();
+    }
+};
+
+int main(int argc, char *argv[]) {
+    // Declare delegate
+    A ^ a = gcnew A;
+    MyDel ^ DelInst = gcnew MyDel(a, &A::capitalize);
+
+    // Cast into Func
+    Func<String ^, String ^> ^ selector = reinterpret_cast<Func<String ^, String ^> ^>(DelInst);

```

Comment:
@mairaw can you check the formatting here? I'm not sure what formatting guidelines do we use in the samples

Reply:
Not sure if we have any for C++. @BillWagner @gewarren?

Reply:
@corob-msft might be the best person to ask about C++ formatting guidelines.

Reply:
Formatting is not a big issue here. I usually paste code into VS for the autoformatting and then copy it back out. It likes to keep hats with their types. I prefer block style to K&R, but we've never mandated it. I'd give the delegate wrapper class a more evocative name than `A`, but it's pretty much a quote of [one of our other examples](https://docs.microsoft.com/en-us/cpp/dotnet/how-to-define-and-use-delegates-cpp-cli?view=msvc-160), so I can't complain too loudly.

Reply:
One thing: I'd remove the parameters to `main()`. They aren't used (which causes warnings at /W4), aren't needed, and in the current /clr console app template, we use `int main(array<System::String ^> ^args)` when we want to refer to the arguments.

Reply:
Thanks for reviewing it! I can 1) remove the parameters to main and 2) change the name `A` to `DelegateWrapper`

Reply:
@coffeeologist One thing I failed to mention earlier: Method names should have an initial capital. Consider renaming `capitalize` to `ToUpper`, since that's what it does. As for reviewers, I don't think I have write access to this repo, so we'd need to get someone on the dotnet team for that.

Reply:
Thanks for the catch; I agree. Made the change!